module/zfs/dsl_scan.c

```diff
@@ -3173,12 +3213,16 @@ dsl_scan_sync(dsl_pool_t *dp, dmu_tx_t *tx)
 	spa_t *spa = dp->dp_spa;
 	state_sync_type_t sync_type = SYNC_OPTIONAL;
 
+	if (spa->spa_resilver_defered &&
+	    !spa_feature_is_active(dp->dp_spa, SPA_FEATURE_RESILVER_DEFER))
+		spa_feature_incr(spa, SPA_FEATURE_RESILVER_DEFER, tx);
+
 	/*
 	 * Check for scn_restart_txg before checking spa_load_state, so
 	 * that we can restart an old-style scan while the pool is being
 	 * imported (see dsl_scan_init).
 	 */
-	if (dsl_scan_restarting(scn, tx)) {
+	if (!spa->spa_resilver_defered && dsl_scan_restarting(scn, tx)) {
```

Comment:
I wonder if it could be better to never set `scn_restart_txg`, rather than setting it and then ignoring it here?

Reply:
We already don't set it. I originally added it to deal with the old-style resilver code that also wanted to set th restart txg. However, this code should never be run when features are active anyway, so it should be safe to put this line back to the way it was.